advanced/wallets/react-wallet-v2/src/lib/smart-accounts/KernelSmartAccountLib.ts

```diff
@@ -279,6 +289,66 @@ export class KernelSmartAccountLib implements EIP155Wallet {
     return serializedSessionKey
   }
 
+  async grantPermissions(
+    grantPermissionsRequestParams: WalletGrantPermissionsParameters
+  ): Promise<WalletGrantPermissionsReturnType> {
+    if (!this.publicClient) {
+      throw new Error('Client not initialized')
+    }
+    console.log('grantPermissions', { grantPermissionsRequestParams })
+
+    const signer = grantPermissionsRequestParams.signer
+    // check if signer type is  AccountSigner then it will have data.id
+    if (signer && !(signer.type === 'key')) {
+      throw Error('Currently only supporting KeySigner Type for permissions')
+    }
+
+    const typedSigner = signer as KeySigner
+    const id = typedSigner.data.id
+    const targetAddress = id.replace('did:ethr:', '')
```

Comment:
will it receive the target Address as did:ethr:{AccountAddress} or did:key:{AccountPublicKey}?

Reply:
It will receive it as `did:ethr:{AccountAddress}`

Reply:
Changed to use standard did key